enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -54,37 +54,24 @@ pub struct Handler<'a> {
 
 impl<'a> Write for Handler<'a> {
     fn write_str(&mut self, s: &str) -> core::fmt::Result {
-        let bytes = s.as_bytes();
-        let len = bytes.len() as _;
-        if len == 0 {
+        if s.as_bytes().len() == 0 {
             return Ok(());
         }
 
-        // Allocate some unencrypted memory.
-        let map = match self.ualloc(len) {
-            Err(_) => return Err(core::fmt::Error),
-            Ok(map) => map,
+        let mut cursor = self.block.cursor();
+        let untrusted = match cursor.copy_slice(s.as_bytes()).or(Err(core::fmt::Error)) {
+            Ok(slice) => slice,
+            Err(e) => return Err(e),
```

Comment:
How about the `?` operator here?

```diff
diff --git a/enarx-keep-sgx-shim/src/handler.rs b/enarx-keep-sgx-shim/src/handler.rs
index b8bc7f1..0be3cbd 100644
--- a/enarx-keep-sgx-shim/src/handler.rs
+++ b/enarx-keep-sgx-shim/src/handler.rs
@@ -59,10 +59,7 @@ impl<'a> Write for Handler<'a> {
         }
 
         let mut cursor = self.block.cursor();
-        let untrusted = match cursor.copy_slice(s.as_bytes()).or(Err(core::fmt::Error)) {
-            Ok(slice) => slice,
-            Err(e) => return Err(e),
-        };
+        let untrusted = cursor.copy_slice(s.as_bytes()).or(Err(core::fmt::Error))?;
 
         // Do the syscall; replace encrypted memory with unencrypted memory.
         let req = request!(libc::SYS_write => libc::STDERR_FILENO, untrusted, s.as_bytes().len());
```

Reply:
@connorkuehl When I had tried to use `?`, it told me it couldn't convert the error type from `copy_slice`, which is `()`, to a `core::fmt::Error` (which is required by the `Write` trait).

Reply:
Did you leave on the `.or(Err(core::fmt::Error))`? That diff in my previous comment compiles for me.

Reply:
Let me give it a shot.

Reply:
You were right! :) I think when I tried it maybe my code was structured differently.